src/20231208_Multi_TreasuryManagementPolygonV2ToV3Migration/AaveV2Polygon_TreasuryManagementPolygonV2ToV3Migration_20231208.t.sol

```diff
@@ -0,0 +1,105 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import {IERC20} from 'solidity-utils/contracts/oz-common/interfaces/IERC20.sol';
+import {AaveV2Polygon, AaveV2PolygonAssets} from 'aave-address-book/AaveV2Polygon.sol';
+import {AaveV3Polygon, AaveV3PolygonAssets} from 'aave-address-book/AaveV3Polygon.sol';
+import {ProtocolV2TestBase} from 'aave-helpers/ProtocolV2TestBase.sol';
+
+import {AddressesToMigrate, AaveV2Polygon_TreasuryManagementPolygonV2ToV3Migration_20231208} from './AaveV2Polygon_TreasuryManagementPolygonV2ToV3Migration_20231208.sol';
+
+/**
+ * @dev Test for AaveV2Polygon_TreasuryManagementPolygonV2ToV3Migration_20231208
+ * command: make test-contract filter=AaveV2Polygon_TreasuryManagementPolygonV2ToV3Migration_20231208
+ */
+contract AaveV2Polygon_TreasuryManagementPolygonV2ToV3Migration_20231208_Test is
+  ProtocolV2TestBase
+{
+  AaveV2Polygon_TreasuryManagementPolygonV2ToV3Migration_20231208 internal proposal;
+
+  function setUp() public {
+    vm.createSelectFork(vm.rpcUrl('polygon'), 50991163);
+    proposal = new AaveV2Polygon_TreasuryManagementPolygonV2ToV3Migration_20231208();
+  }
+
+  function test_execute() public {
+    // Collector contract is same address across both files
+    assertEq(address(AaveV2Polygon.COLLECTOR), address(AaveV3Polygon.COLLECTOR));
+
+    address[8] memory TO_MIGRATE = AddressesToMigrate.getUnderlyingAddresses();
+    address[8] memory A_TOKENS = AddressesToMigrate.getV2ATokenAddresses();
+
+    address[8] memory V3_A_TOKENS = [
+      AaveV3PolygonAssets.USDC_A_TOKEN,
+      AaveV3PolygonAssets.DAI_A_TOKEN,
+      AaveV3PolygonAssets.USDT_A_TOKEN,
+      AaveV3PolygonAssets.WETH_A_TOKEN,
+      AaveV3PolygonAssets.WMATIC_A_TOKEN,
+      AaveV3PolygonAssets.WBTC_A_TOKEN,
+      AaveV3PolygonAssets.GHST_A_TOKEN,
+      AaveV3PolygonAssets.LINK_A_TOKEN
+    ];
+
+    uint256[] memory v3AssetsBalances = new uint256[](8);
+    uint256[] memory v2Assetsbalances = new uint256[](8);
+
+    for (uint256 i = 0; i < TO_MIGRATE.length; ++i) {
+      uint256 balanceV2 = IERC20(A_TOKENS[i]).balanceOf(address(AaveV2Polygon.COLLECTOR));
+      uint256 balanceV3 = IERC20(V3_A_TOKENS[i]).balanceOf(address(AaveV3Polygon.COLLECTOR));
```

Comment:
It works, but it is better to use `AaveV3Polygon.COLLECTOR` always, instead of mixing up v2 and v3, because they are the same address.
As in this case the objective is to check balances' consistency post-proposal, to be double sure you can just do an asset that both collectors are the same address.

This also applies to the payload, where you use sometimes v2 and others v3 for what it seems arbitrary reasons